week-two/DoubleLinkedList.java

```diff
@@ -0,0 +1,76 @@
+public class DoubleLinkedList {
+  public DoubleNode head;
+
+  public DoubleLinkedList() {
+    this.head = null;
+  }
+
+  public void add(String z) {
+    DoubleNode node = new DoubleNode(z);
+    DoubleNode last = this.head;
+    node.next = null;
+    if (head != null) {
+      while (last.next != null) {
+        last = last.next;
```

Comment:
I see, so `last` is really your temporary node or the iterator//cursor object. Reading L10 was initially confusing since I was wondering why last was called head.